### PR TITLE
Hotfix 2.2.6

### PR DESCRIPTION
**Pull request by DavidBoike**
Connects to Particular/PlatformDevelopment#697

Contains fixes for:

* #142 Remove the server version check
* #177 FindTypeTagName conventions change if method of creating DocumentStore changes

@Particular/persistence-maintainers please review for release

### CODE REVIEW

File: src/NServiceBus.RavenDB.Tests/Infrastructure/ReusableDB.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            }
+
+            var docStore = new DocumentStore
+            {
+                Url = "http://localhost:8081"
+            };
+
+            docStore.Initialize();
+
+            for (var i = 0; i < 4; i++)
```
Comment (HEskandari) [inline src/NServiceBus.RavenDB.Tests/Infrastructure/ReusableDB.cs]: Might be nitpicking being a 'test' code and all but could we write this more functionally? The intent would be expressed better when not mixed with the retry logic. http://stackoverflow.com/questions/1563191/c-sharp-cleanest-way-to-write-retry-logic

**Comment by HEskandari** [comment]
Had a look, seem to be good. Although a second review wouldn't hurt. @colin-higgins

File: src/NServiceBus.RavenDB/Internal/DocumentIdConventions.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    using Raven.Client;
+    using Raven.Json.Linq;
+
+    class DocumentIdConventions
+    {
+        private readonly IDocumentStore store;
+        private readonly Func<Type, string> userSuppliedConventions;
+        private readonly string endpointName;
+        private readonly string collectionNamesDocId;
+        private readonly object padlock;
```
Comment (colin-higgins) [inline src/NServiceBus.RavenDB/Internal/DocumentIdConventions.cs]: I like the name :)

**Comment by colin-higgins** [comment]
I'm good with it as well.

**Pull request merged**